shared/src/test/scala/squants/QuantitySpec.scala

```diff
@@ -605,4 +606,81 @@ class QuantitySpec extends FlatSpec with Matchers with CustomMatchers {
     val ts = List(Thangs(1000), Kilothangs(10), Kilothangs(100))
     ts.sum should be(Kilothangs(111))
   }
+
+  behavior of "Dimension"
+
+  it should "Parse a String into a Quantity based on the supplied Type parameter" in {
+    import squants.mass.Mass
+    import squants.space.Length
+    import squants.time.Time
+
+    def parse[A <: Quantity[A]: Dimension](s: String): Try[A] = {
+      implicitly[Dimension[A]].parseString(s)
+    }
+
+    implicit val length = Length
+    implicit val time = Time
+    implicit val thingee = Thingee
+    implicit val mass = Mass
+
+    val l = parse[Length]("100 m")
+    val t = parse[Time]("100 m")
+    val th = parse[Thingee]("100 th")
+    val m = parse[Mass]("100 m")
+
+    l.get should be(Meters(100))
+    t.get should be(Minutes(100))
+    th.get should be(Thangs(100))
+    m.isFailure should be(true)
+  }
+
+  it should "Parse a Tuple with a Double into a Quantity based on the supplied Type parameter" in {
+    import squants.mass.Mass
+    import squants.space.Length
+    import squants.time.Time
+
+    def parse[A <: Quantity[A]: Dimension](t: (Double,  String)): Try[A] = {
+      implicitly[Dimension[A]].parseTuple[Double](t)
+    }
+
+    implicit val length = Length
+    implicit val time = Time
+    implicit val thingee = Thingee
+    implicit val mass = Mass
+
+    val l = parse[Length]((100d, "m"))
+    val t = parse[Time]((100d, "m"))
+    val th = parse[Thingee]((100d, "th"))
+    val m = parse[Mass]((100d, "m"))
+
+    l.get should be(Meters(100d))
```

Comment:
rather than calling `.get()` on the `Try`s, I think the code would look cleaner with ScalaTest's `TryValues` - http://doc.scalatest.org/2.2.6/#org.scalatest.TryValues.

There's an example (using `EitherValues`) in the docs - http://www.scalatest.org/user_guide/other_goodies#eitherValues. It's straightforward to convert to `TryValues`.